client/host_queue.go

```diff
@@ -584,3 +578,105 @@ func errQueueUnknownOperation(hostID string) error {
 func errQueueFetchNoResponse(hostID string) error {
 	return fmt.Errorf("host operation queue did not receive response for given fetch for host: %s", hostID)
 }
+
+// ops container types
+
+type namespaceWriteBatchOps struct {
```

Comment:
opened https://github.com/m3db/m3db/issues/531 to track de-duplication. mind putting a `// TODO: use genny to make namespaceWriteBatchOps and namespaceWriteTaggedBatchOps share code (https://github.com/m3db/m3db/issues/531)` and elsewhere as you find applicable

Reply:
Sure thing.